includes/class-wp-job-manager-recaptcha.php

```diff
@@ -0,0 +1,242 @@
+	public function __construct() {
+		$this->site_key          = get_option( 'job_manager_recaptcha_site_key' );
+		$this->secret_key        = get_option( 'job_manager_recaptcha_secret_key' );
+		$this->recaptcha_version = get_option( 'job_manager_recaptcha_version' );
+
+		if ( $this->use_recaptcha_field() ) {
+			add_action( 'submit_job_form_end', [ $this, 'display_recaptcha_field' ] );
+			add_filter( 'submit_job_form_validate_fields', [ $this, 'validate_recaptcha_field' ] );
+			add_filter( 'submit_draft_job_form_validate_fields', [ $this, 'validate_recaptcha_field' ] );
+		}
+	}
+
+	/**
+	 * Returns static instance of class.
+	 *
+	 * @return self Main instance.
+	 */
+	public static function instance() {
+		if ( is_null( self::$instance ) ) {
+			self::$instance = new self();
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Use reCAPTCHA field on the form?
+	 *
+	 * @return bool
+	 */
+	public function use_recaptcha_field() {
+		if ( ! $this->is_recaptcha_available() ) {
+			return false;
+		}
+		return 1 === absint( get_option( 'job_manager_enable_recaptcha_job_submission' ) );
+	}
+
+	/**
+	 * Enqueue the scripts for the form.
+	 */
+	public static function enqueue_scripts() {
+		$instance = self::instance();
+
+		if (
+			$instance->use_recaptcha_field() &&
+			in_array( $instance->recaptcha_version, [ 'v2', 'v3' ], true )
+		) {
+			$recaptcha_version = $instance->recaptcha_version;
+			$recaptcha_url     = '';
+
+			if ( 'v2' === $recaptcha_version ) {
+				$recaptcha_url = 'https://www.google.com/recaptcha/api.js';
+			} elseif ( 'v3' === $recaptcha_version ) {
+				$recaptcha_url = 'https://www.google.com/recaptcha/api.js?render=' . $instance->site_key;
+			}
+			wp_enqueue_script( 'recaptcha', $recaptcha_url, [], JOB_MANAGER_VERSION, [ 'strategy' => 'defer' ] );
+		}
+	}
+
+	/**
+	 * Checks whether reCAPTCHA has been set up and is available.
+	 *
+	 * @return bool
+	 */
+	public function is_recaptcha_available() {
+		$is_recaptcha_available = ! empty( $this->site_key ) && ! empty( $this->secret_key );
+
+		/**
+		 * Filter whether reCAPTCHA should be available for this form.
+		 *
+		 * @since 1.30.0
+		 *
+		 * @param bool $is_recaptcha_available
+		 */
+		return apply_filters( 'job_manager_is_recaptcha_available', $is_recaptcha_available );
+	}
+
+	/**
+	 * Dispaly the reCAPTCHA field in the form.
+	 *
+	 * @return void
+	 */
+	public function display_recaptcha_field() {
+		$field             = [];
+		$field['label']    = get_option( 'job_manager_recaptcha_label' );
+		$field['required'] = true;
+		$field['site_key'] = $this->site_key;
+
+		if ( 'v2' === $this->recaptcha_version ) {
+			get_job_manager_template(
+				'form-fields/recaptcha-field.php',
+				[
+					'key'   => 'recaptcha',
+					'field' => $field,
+				]
+			);
+		} elseif ( 'v3' === $this->recaptcha_version ) {
+			get_job_manager_template(
+				'form-fields/recaptcha-v3-field.php',
+				[
+					'key'   => 'recaptcha',
+					'field' => $field,
+				]
+			);
+		}
```

Comment:
Given that the parameters to the teemplate are the same, WDYT about changing this logic to just change the template filename?

So we would have just one call to `get_job_manager_template` and the `if` / `elseif` to decide on the filename to load..

Reply:
Good idea!  done in f6aea0315466075b6e9d99e332460c54b33b062d